src/core/src/Environment.C

```diff
@@ -1782,9 +1759,17 @@ void queso_terminate_handler()
 
 
 //-------------------------------------------------------
+// Previous behavior:
+//   * IF      alternative options passed in, use them exclusively
+//   * ELSE IF input file, parse and set options from it
+//   * ELSE    use default options
+// New behavior:
+//   * START   with default options or passed alternative options
+//   * IF      input file, any parsed options override stored values
```

Comment:
I'd put this comment in the header, where many people will read it as they use the API, rather than in the source file, where nobody but core developers will look unless they're already tearing their hair out over failure to understand the behavior.